jupyterhub/templates/hub/netpol-singleuser.yaml

```diff
@@ -1,35 +1,35 @@
-{{- if and .Values.singleuser.networkPolicy.enabled }}
+{{- if and .Values.singleuser.networkPolicy.enabled -}}
 apiVersion: networking.k8s.io/v1
 kind: NetworkPolicy
 metadata:
   name: singleuser-network-policy
+  labels:
+    {{- include "jupyterhub.labels" . | nindent 4 }}
 spec:
   podSelector:
     matchLabels:
-      app: jupyterhub
-      component: singleuser-server
+      {{- $_ := merge (dict "componentLabel" "singleuser-server") . }}
```

Comment:
Can you expand a bit on what this is doing?

Reply:
Yes check out the text I added to https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/625#issuecomment-378071689 under "Passing merged scopes to templates", it might sometimes be overkill, but often it is a nice practice.

For example, when we had the daemonset template and passed a dictionary with two keys, one key was called "name" and one was called "top", with this technique, you could pass a scope that merges the name dict into the current scope reducing the need to utilize .top.Release.Name and similar.

Reply:
Thank you for reviewing @yuvipanda I find it really motivating :D